src/device.h

```diff
@@ -78,6 +78,8 @@ namespace librealsense
 
         virtual bool contradicts(const stream_profile_interface* a, const std::vector<stream_profile>& others) const override;
         virtual double get_device_time(); // Returns time in miliseconds.
+        virtual void start_time_keeper() {};
```

Comment:
I would suggest to refactor  `start/stop_time_keeper` into `enable_time_keeper(bool)`.
The `start/stop_time_keeper` operations are internal to the device's flow (i.e. bound to the start/stop device operations implicitly) and shall be not exposed in the class public API.
On the other hand - `enable_time_keeper` will be connected to the user's dev.set_option(...) operation